rhcos/jobspec.yaml

```
@@ -0,0 +1,102 @@
+      # This repo provides 'qemu-kiwi' for the `sandboxed-containers` extension
+      # https://github.com/openshift/enhancements/blob/master/enhancements/sandboxed-containers/sandboxed-containers-tech-preview.md
+
+      [rhel-8-advanced-virt]
+      baseurl=http://rhsm-pulp.corp.redhat.com/content/dist/layered/rhel8/$basearch/advanced-virt/os/
+      enabled=1
+      gpgcheck=0
+      [rhel-8-fast-datapath]
+      baseurl=http://rhsm-pulp.corp.redhat.com/content/dist/layered/rhel8/$basearch/fast-datapath/os/
+      enabled=1
+      gpgcheck=0
+
+      # Fast Datapath has yet to ship for ARM, so the above repo is empty on aarch64.
+      # Once it starts shipping, the following repo should be removed.
+      [rhel-8-fast-datapath-aarch64]
+      baseurl=http://download-node-02.eng.bos.redhat.com/rcm-guest/puddles/Multi-Arch/Fast-DataPath/EL8-Production/latest/aarch64/os/
+      enabled=1
+      gpgcheck=0
+      # The baseurls below are tracking the latest version of RHEL that has been released.
+      # In the case of any of these locations being unavailable, it should be possible
+      # to switch out the `/8/` to the latest RHEL version (i.e. `/8.1/` or `/8.2/`).
+      #
+      # RHCOS 4.8 will track RHEL 8.4, but we can use `8` until 8.4 reaches EUS
+      [rhel-8-baseos]
+      baseurl=http://rhsm-pulp.corp.redhat.com/content/dist/rhel8/8/$basearch/baseos/os/
+      enabled=1
+      gpgcheck=0
+
+      [rhel-8-appstream]
+      baseurl=http://rhsm-pulp.corp.redhat.com/content/dist/rhel8/8/$basearch/appstream/os/
+      enabled=1
+      gpgcheck=0
+      # Exclude toolbox to make sure we use coreos/toolbox from rhaos repo
+      # kata-containers wants non-modular libpmem
+      exclude=toolbox libpmem*module*
+
+      [rhel-8-nfv]
+      baseurl=http://rhsm-pulp.corp.redhat.com/content/dist/rhel8/8/x86_64/nfv/os/
+      enabled=1
+
+publish_oscontainer:
+  buildstrategy_tls_verify: null
+  registries:
+  - url: quay.io/openshift-release-dev/ocp-v4.0-art-dev
+    secret_type: cluster
+    secret: upload-registry.ci.openshift.org.dockercfg
+  - url: registry.ci.openshift.org/rhcos/machine-os-content:test
+    # should be :4.y
+    secret_type: cluster
+    secret: upload-registry.ci.openshift.org.dockercfg
+
+stages:
+- id: Build and test QEMU
+  description: Create the base build and test it
+  build_artifacts: [base]
+  post_commands:
+    - cosa kola run --basic-qemu-scenarios
+  execution_order: 1
+- id: Upload to brew
+  description: Create brew build record and upload artifacts
+  require_artifacts: [qemu]
+  commands:
+    - cosa ... ?
+      # secret is rhcos-keytab
+      # can this retry on failure?
+  execution_order: 2
+- id: Upload to S3
+  description: Insert into S3 (but not in standard stream location)
+  require_artifacts: [qemu]
+  commands:
+    - cosa buildupload s3 --acl=private art-rhcos-ci/testing/gangplank
+      # secret is coreos-assembler-aws-secret
+  execution_order: 2
```

```suggestion
execution_order: 3
```
It will probably change to 4 after adding the buildprep